addons/mail/static/tests/chatter_tests.js

```diff
@@ -520,6 +520,101 @@ QUnit.test('attachmentBox basic rendering', async function (assert) {
     form.destroy();
 });
 
+QUnit.test('attachmentBox drag and drop files [REQUIRE NON-INCOGNITO WINDOW]', async function (assert) {
+    assert.expect(12);
+    this.data.partner.records.push({
+        id: 7,
+        display_name: "attachment_test",
+    });
+
+    testUtils.mock.patch(AttachmentBox, {
+        _callUploadAttachment: function (controllerUrl, formData) {
+            assert.strictEqual(controllerUrl, "/web/binary/upload_attachment",
+                "upload url should match to /web/binary/upload_attachment");
+            assert.strictEqual(formData.get("ufile").name, "text.txt",
+                "uploading file name should be 'text.txt'");
+            this.trigger_up('reload_attachment_box');
+            return Promise.resolve();
+        }
+    });
+
+    var form = await createView({
+        View: FormView,
+        model: 'partner',
+        data: this.data,
+        services: this.services,
+        arch: '<form string="Partners">' +
+                '<sheet>' +
+                    '<field name="foo"/>' +
+                '</sheet>' +
+                '<div class="oe_chatter">' +
+                    '<field name="message_ids" widget="mail_thread"/>' +
+                '</div>' +
+            '</form>',
+        res_id: 7,
+        mockRPC: function (route, args) {
+            var result = this._super.apply(this, arguments);
+            if (args.method === "search_read" && args.model === "ir.attachment") {
+                return result.then(function (records) {
+                    return Promise.resolve(records.concat([{
+                        id: 3,
+                        type: 'binary',
+                        mimetype: "text/plain",
+                        name: "text.txt",
+                        res_id: 7,
+                        res_model: 'partner'
+                    }]));
+                });
+            }
+            return result;
+        }
+    });
+
+    var $button = form.$('.o_chatter_button_attachment');
+    assert.strictEqual($button.length, 1, "should have one attachment button");
+    await testUtils.dom.click($button);
+
```

Comment:
white spaces do not help readability of test.
I suggest grouping successive actions+direct assertions together, example:

```js
const form = await createView();
assert.strictEqual();

await testUtils.dom.click();
assert.containsOnce();
assert.isNotVisible();

await testUtils.dom.click();
await testUtils.nextTick();
assert.containsNone();

const file = await testUtils.file.createFile();
await testUtils.file.dragoverFile();
await testUtils.nextTick();
assert.containsOnce();
```

Reply:
Done